### ISSUE TITLE
Default wallpaper shown instead of custom one in workspace view

### ISSUE THREAD

**Issue by BLovegrove** [pop-os/pop#2069]
**Distribution:**
NAME="Pop!_OS"
VERSION="21.10"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 21.10"
VERSION_ID="21.10"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=impish
UBUNTU_CODENAME=impish
LOGO=distributor-logo-pop-os

**Issue/Bug Description:**
Default background is shown when in workspace view (both via super key and hot-corner).
Before:
![image](https://user-images.githubusercontent.com/32378871/146249364-0411f5df-136d-469b-b554-5f43e2fe66dc.png)

After:
![image](https://user-images.githubusercontent.com/32378871/146249478-3cd1ca2c-d41b-41a1-b1c3-ba1ceec5a23c.png)

**Steps to reproduce (if you know):**
Just open workspace view in any way after changing your desktop background.

**Expected behavior:**
My actual desktop background be shown as the background image.

**Comment by leviport** [comment]
I've seen this previously when investigating https://github.com/pop-os/beta/issues/230, however it seems like it comes and goes. When it's in a state where this is happening, does restarting Gnome Shell with Alt+F2, `r`, Enter make the proper wallpaper appear in the workspaces overview?

**Comment by garybowers** [comment]
I think this is linked to the issue I raised .... when selecting a wallpaper, selecting workspaces shows the new paper and changing anything on the display (zoom or resolution) appears to force it to apply.

**Comment by BLovegrove** [comment]
Restarting Gnome Shell with Alt+F2, ```r```, Enter fixes the problem. Would also like to note that the default gnome activities screen doesn't exhibit this issue.

**Comment by raf-r** [comment]
This happens to me too. Changing the wallpaper in the settings does not change the actual image in the desktop. Restarting gnome does solve the issue.

**Comment by jabbermacy** [comment]
> I've seen this previously when investigating [pop-os/beta#230](https://github.com/pop-os/beta/issues/230), however it seems like it comes and goes. When it's in a state where this is happening, does restarting Gnome Shell with Alt+F2, `r`, Enter make the proper wallpaper appear in the workspaces overview?

YES this does allow showing correct wallpaper (I have exact same issue)

**Comment by PHY1643** [comment]
Pop!_OS 21.10 on a Librem 14.

For what it's worth: I too have experienced this, but _only_ when running Pop! on X. If I login using Wayland, I never see this issue. (Though, using Wayland introduces some other minor issues.)  Regardless, restarting GNOME shell temporarily fixes the issue.
